### DISCUSSION TITLE
How to get the messages inside the queue

### DISCUSSION THREAD

**Discussion by alvenkatesh** [rabbitmq/rabbitmq-website#1927] [Q&A] [answered]
Hi all,

With the help of producers, we will put messages into the queue, and with consumers, we will retrieve messages from the queue. What do I need to do if I want to retrieve the list of messages currently inside the queue?

**Comment by jayshah1819** [comment]
I guess Rabbitmq doesn't have a direct method to list all messages inside a queue, but you can check the status.

**Comment by michaelklishin** [comment, marked as answer]
Messages in a stream can be consumed continuously. Messages in a queue can be consumed and requeued but that will affect queue metadata and, depending on the concurrent activity, can affect total message ordering.

**Comment by jayshah1819** [comment]
Just to confirm for queues, the only way to inspect messages is by actually consuming them  (via basic. get or a consumer), and then optionally requeuing them. But doing so can impact queue metadata and message ordering, especially under concurrent load , is that right?

In that context, I had a follow-up question
 For non-production environments, is using the RabbitMQ Management HTTP API (e.g., POST /api/queues/{vhost}/{name}/get with "requeue": true) considered a safe and supported method to peek at messages?

Or is there a more recommended alternative to inspect the contents of a queue without disrupting normal operations, I'm currently working on debugging and monitoring, not in a production environment.
Thanks in advance! @michaelklishin

**Reply by michaelklishin** [reply]
@jayshah1819 any way of consuming and requeueing will have exactly the same impact on the message metadata, and potential impact on the message's relative position within a queue.
